### PR TITLE
index category with search

### PR DESCRIPTION
**Pull request by leduykhanhhedspi**

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -3,4 +3,14 @@ def full_title page_title = ""
     base_title = t "static_pages.framgia"
     page_title.empty? ? base_title : page_title + " | " + base_title
   end
+
+  def link_to_add_fields(name, f, type)
```
Comment (VuThiHien94) [inline app/helpers/application_helper.rb]: bo ()

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -3,4 +3,14 @@ def full_title page_title = ""
...
+    new_object = f.object.send "build_#{type}"
+  	id = "new_#{type}"
```
Comment (duccm94) [inline app/helpers/application_helper.rb]: indent

File: app/views/admin/categories/_condition_fields.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    </div>
+    <div class="col-md-4">
+      <%= f.predicate_select %>
+    </div>
+    <div class="col-md-4">
+      <%= f.value_fields do |v| %>
+        <%= v.text_field :value, class: "form-control" %>
+      <% end %>
+    </div> 
+    <%= link_to t("remove"), '#', id: "remove_fields", class: "btn btn-warning" %>
```
Comment (duccm94) [inline app/views/admin/categories/_condition_fields.html.erb]: " instead of '

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -3,4 +3,14 @@ def full_title page_title = ""
...
+  	fields = f.send("#{type}_fields", new_object, child_index: id) do |builder|
```
Comment (VuThiHien94) [inline app/helpers/application_helper.rb]: bo ()

File: app/views/admin/categories/index.html.erb [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
-            class: "btn btn-danger" %>
-        </td>
-      </tr>
-    </table>
-  </div>
+<%= search_form_for @search, url: admin_categories_path do |f| %>
+  <%= f.condition_fields do |c| %>
+    <%= render "condition_fields", f: c %>
+  <% end %>
+  <div><%= link_to_add_fields "Add Conditions", f, :condition %></div>
```
Comment (duccm94) [inline app/views/admin/categories/index.html.erb]: I18n

File: app/helpers/application_helper.rb [hunk 6]
```diff
@@ -3,4 +3,14 @@ def full_title page_title = ""
...
+      render(type.to_s + "_fields", f: builder)
```
Comment (VuThiHien94) [inline app/helpers/application_helper.rb]: bo ()

File: app/helpers/application_helper.rb [hunk 7]
```diff
@@ -3,4 +3,14 @@ def full_title page_title = ""
...
+    end
+    link_to(name, "#", class: "add_fields",
```
Comment (VuThiHien94) [inline app/helpers/application_helper.rb]: bo ()

File: config/routes.rb [hunk 8]
```diff
@@ -3,7 +3,7 @@
   root "static_pages#home"
   get "about" => "static_pages#about"
   namespace :admin do
-    resources :categories, except: [:show]
+    resources :categories
   end 
   resources :users, only: [:index, :show] do
     resources :relationships, only: [:index]
```
Comment (duccm94) [inline config/routes.rb]: bỏ []

File: app/controllers/admin/categories_controller.rb [hunk 9]
```diff
@@ -6,8 +6,10 @@ def new
 
   def edit
   end
-  
+
   def index
+    @search = Category.search params[:q]
```
Comment (duccm94) [inline app/controllers/admin/categories_controller.rb]: categories.search

**Comment by VuThiHien94** [comment]
ok

**Comment by duccm94** [comment]
OK

**Pull request merged**